File: Assignment 2/a2q2.java
/*Regan Ross, 0369345
COSC 1046, Dr. Aaron Langille*/

import java.util.Scanner;

public class StringChecker {
	public static void main(String []args) {
		
		Scanner input = new Scanner(System.in);
		
		System.out.println("Enter 3 strings, one per line:");
		
			String s1 = input.nextLine();
			String s2 = input.nextLine();
			String s3 = input.nextLine();
			
	
			
				int result1 = (s1.compareTo(s2));
				int result2 = (s2.compareTo(s3));
			
			
			if ((s1.equals(s2)) && (s2.equals(s3)))	
					System.out.println("The strings are equal.");
			
			else if ((result1 <= 0) && (result2 <= 0))
				System.out.println("Your strings are in lexographical order (ascending) !");
			
			else if ((result1>= 0) && (result2 >= 0))
					System.out.println("Your strings are in lexographical order (descending) !");
			else
				System.out.println("Those strings are not in any order. :(");
			
	}
}

File: Assignment 2/cpp/a2q2.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main(){
    string words[3];

    cout << "Enter three words:" << endl;
    cin >> words[0] >> words[1] >> words[2];

    bool ascending = true;
    bool descending = true;
    bool equivalent = true;

    for(int i = 0; i < 2; i++){
        if (words[i] != words[i + 1])
        {
            equivalent = false;
        }
        if (words[i+1] > words[i])
        {
            descending = false;
        }
        if (words[i] > words[i+1])
        {
            ascending = false;
        }
    }

    if (equivalent)
    {
        cout << "Your words are all equal!" << endl;
    }
    else if (descending)
    {
        cout << "Your words are in reverse lexicographical order!" << endl;
    }
    else if (ascending)
    {
        cout << "Your words are in lexicographical order!" << endl;
    }

}